### PR TITLE
Beacons are no longer radios

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 

:cl: Dax Dupont
add: Beacons can now be toggled on and off.
add: Mappers can now have beacons that default to off. Useful for ruins!
tweak: Renaming replaces the snowflake locator frequency/code
refactor: Beacons are no longer radios. Why were they radios in the first place? I don't know.
/:cl:

[why]: I wanted to add spawning off but I decided to redo all of the things instead.

### CODE REVIEW

**Comment by epochayur** [comment]
RIP ghetto radios

**Comment by DaxDupont** [comment]
@epochayur they didn't work as radios though? There was a return on hear.

**Comment by epochayur** [comment]
woops guess i've got brain damage

**Comment by PKPenguin321** [comment]
You could use them by using :l or :r if they were in your left or right hand, respectively. Elder knowledge tier thing that was useful to me exactly once ever

**Comment by ExcessiveUseOfCobblestone** [comment]
can you use :i on them?

**Comment by DaxDupont** [comment]
what the fuck is wrong with old code, this is almost as bad as clown shoe squeak code being in turfs

**Comment by ShizCalev** [comment]
conflict.

**Comment by DaxDupont** [comment]
Don't merge yet, still fixing the conflict properly

**Comment by DaxDupont** [comment]
done

File: code/game/objects/items/devices/beacon.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		GLOB.teleportbeacons += src
+	else 
+		icon_state = "beacon-off"
+		GLOB.teleportbeacons.Remove(src)
+	to_chat(user, "<span class='notice'>You [enabled ? "enable" : "disable"] the beacon.</span>")
+	return
+
+/obj/item/device/beacon/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/pen)) // needed for things that use custom names like the locator
+		var/new_name = stripped_input(user, "What would you like the name to be?")
```
Comment (ShizCalev) [inline code/game/objects/items/devices/beacon.dm]: add underneath the input
```
if(!user.canUseTopic(src, BE_CLOSE))
return
```

**Review by ShizCalev — DISMISSED** [review]

File: code/game/objects/items/devices/beacon.dm (hunk 1 above)
Comment (DaxDupont) [reply code/game/objects/items/devices/beacon.dm]: @ShizCalev done

**Pull request merged**